ESSArch_Core/frontend/static/frontend/scripts/services/filters.ts

```diff
@@ -0,0 +1,370 @@
+import * as angular from 'angular';
+
+interface rootscope extends ng.IRootScopeService {
+  auth: any;
+}
+
+interface templateOptions {
+  label: string;
+  options?: any[] | Function;
+  optionsFunction?: Function;
+  valueProp?: string;
+  labelProp?: string;
+  trueValue?: any;
+  falseValue?: any;
+  appendToBody?: boolean;
+  refresh?: Function;
+  clearEnabled?: boolean;
+}
+
+interface formlyField {
+  key: string;
+  type: string;
+  templateOptions: templateOptions;
+  defaultValue?: any;
+}
+
+interface formGroup {
+  fieldGroup: formlyField[];
+  className?: string;
+}
```

Comment:
We should try to use `AngularFormly.IFieldObject`, `AngularFormly.IFieldArray`, `AngularFormly.IFieldGroup`, etc. instead of defining our own.